calico/felix/endpoint.py

```diff
@@ -130,6 +132,9 @@ def on_endpoint_update(self, endpoint_id, endpoint):
             self.endpoint_id_by_iface_name[endpoint["name"]] = endpoint_id
 
         if endpoint and endpoint["host"] == self.config.HOSTNAME:
+        # MD4: Do we really need to do all the above work for non-local
+        # endpoints?  Since the config for hostname is always available, can't
+        # we move this check to the top?
```

Comment:
We do need to store non-local endpoints and we need to handle endpoint == None (hence no host).  We should just pass the host in, I think we need to do that anyway for other reasons (it helps us handle any issues with duplicate endpoint IDs during a migration)